plugins/urls/__init__.py

```diff
@@ -178,113 +219,139 @@ def message_handler(self, event=MessageReceived):
-                        return
+            if _url is None:
+                continue
 
-                    if domain is not None and "/" in domain:
-                        domain = domain.split("/")[0]
-                    if domain is None:
-                        target.respond(title)
-                    else:
-                        target.respond("\"%s\" at %s" % (title, domain))
-                elif isinstance(target, User):
-                    if title is None:
-                        return
+            # Check redirects, following as necessary
+
+            redirects = 0
+            max_redirects = self.config.get("redirects", {}).get("max", 15)
+            domains = self.config.get("redirects", {}).get("domains", [])
+
+            self.logger.debug("Checking redirects...")
+
+            while _url.domain in domains and redirects < max_redirects:
+                redirects += 1
+
+                session = Session()
+
+                #: :type: requests.Response
+                r = yield session.get(str(_url), allow_redirects=False)
+
+                if r.is_redirect:
+                    # This only ever happens when we have a well-formed
+                    # redirect that could have been handled automatically
+
+                    redirect_url = r.headers["location"]
+
+                    self.logger.debug(
+                        "Redirect [{0:03d}] {1}".format(
+                            redirects, redirect_url
+                        )
+                    )
+
+                    _url = self.match_to_url(extract_urls(redirect_url)[0])
+
+            lazy_request = LazyRequest(req_args=[str(_url)])
+
+            self.channels.get(protocol.name, {}) \
+                .get(source.name, {})["last"] = str(_url)
+
+            yield self.run_handlers(_url, {
+                "event": event,
+                "config": self.config,
+                "get_request": lazy_request,
+                "redirects": redirects,
+                "max_redirects": max_redirects
+            })
 
-                    if domain is not None and "/" in domain:
-                        domain = domain.split("/")[0]
-                    if domain is None:
-                        source.respond(title)
+    def match_to_url(self, match):
+        """
+        :rtype: URL
+        """
+        # Expand the match to make it easier to work with
+
+        # Input: ''http://x:y@tools.ietf.org:80/html/rfc1149''
+        # Match: '', http, x:y, tools.ietf.org, :80, /html/rfc1149''
+
+        _prefix, _protocol, _basic, _domain, _port, _path = (
+            to_unicode(x) for x in match
+        )
+
+        _port = _port.lstrip(":")  # Remove this as the regex captures it
+
+        try:
+            if _port:
+                _port = int(_port)
+        except ValueError:
+            self.logger.warn("Invalid port: {0}", _port)
+            return None
+
+        _translated = self.translate_prefix(_prefix)
```

Comment:
The prefix removal section seems like it could be split out into its own function to make it easier to test. Then again, this plugin has no tests, so... ¯\(°_o)/¯